crates/rspack_binding_api/src/native_watcher.rs

```diff
@@ -0,0 +1,184 @@
+use std::boxed::Box;
+
+use async_trait::async_trait;
+use napi::bindgen_prelude::FnArgs;
+use napi_derive::*;
+use rspack_fs::{FsWatcher, FsWatcherOptions, Ignored, PathUpdater};
+use rspack_napi::threadsafe_function::ThreadsafeFunction;
+
+struct SafetyIgnored {
+  f: ThreadsafeFunction<String, bool>,
+}
+
+#[async_trait]
+impl Ignored for SafetyIgnored {
+  async fn ignore(&self, path: &str) -> bool {
+    self
+      .f
+      .call_with_sync(path.to_string())
+      .await
+      .unwrap_or_default()
```

Comment:
we shouldn't panic when f call with error

Reply:
Updated.